example/scripts/tests/test-mutex.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const test = require('ava');
+const Mutex = require('../lib/mutex');
+
+test.beforeEach((t) => {
+  t.context.sha = 'someSha';
+  t.context.key = 'testKey';
+  t.context.timeout = 123;
+  t.context.tableName = 'sometable';
+  t.context.getResults = {
+    Item: { sha: t.context.sha, key: t.context.key, expire: t.context.timeout }
+  };
+  t.context.docClient = {
+    get: () => ({ promise: () => t.context.getResults }),
+    put: (params) => ({ promise: () => Promise.resolve(params) }),
+    delete: () => ({ promise: () => true })
+  };
+  t.context.mutex = new Mutex(t.context.docClient, t.context.tableName);
+});
+
+test('Mutex passes correct params to dynamo docClient on writeLock', async (t) => {
+  const key = t.context.key;
+  const timeout = t.context.timeout;
+  const sha = t.context.sha;
+
+  const writeParams = {
+    TableName: t.context.tableName,
+    Item: {
+      key: key,
+      expire: 500 + timeout,
+      sha: sha
+    },
+    ConditionExpression: '#key <> :key OR (#key = :key AND #expire < :expire)',
+    ExpressionAttributeNames: {
+      '#key': 'key',
+      '#expire': 'expire'
+    },
+    ExpressionAttributeValues: {
+      ':key': key,
+      ':expire': 500
+    }
+  };
+
+  const mutex = t.context.mutex;
+  const result = await mutex.writeLock(key, timeout, sha);
+  writeParams.Item.expire = result.Item.expire;
+  writeParams.ExpressionAttributeValues[':expire'] = result.ExpressionAttributeValues[':expire'];
+  t.deepEqual(result, writeParams);
+});
+
+test('Mutex unlock passes corect params to dynamo docclient on unlock', async (t) => {
+  const key = t.context.key;
+  const gitSHA = t.context.sha;
+  const mutex = t.context.mutex;
+
+  const result = await mutex.unlock(key, gitSHA);
```

Comment:
I'm not sure I totally follow what this is testing. We're mocking the docClient to always return true for the delete, so what does the test prove?

Reply:
@markdboyd  The only actual logic in our code in the success case, which is that the method *returns* the value.      The name could be better, however.

Reply:
Ah, indeed. Makes sense

Reply:
Re: naming for class, I'm fond of using this pattern:

`Mutex.unlock() returns true when unlock operation is successful`

Putting the class/method/function name first makes it clear what is being tested and then you specify the test case

Reply:
I can get behind that : 53e70117e